cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsIDBrokerObjectStorageValidator.java

```diff
@@ -0,0 +1,302 @@
+                    .filter(m -> users.contains(m.getKey()))
+                    .map(Map.Entry::getValue).collect(Collectors.toCollection(TreeSet::new));
+    }
+
+    private Set<String> checkRolePermissions(Collection<Role> roles, List<String> policyFileNames,
+                                        String datalakeBucket, String storageLocationBase,
+                                        String s3GuardDynamoDBTableName) {
+        Set<String> failedActions = new TreeSet<>();
+        for (String policyFileName : policyFileNames) {
+            try {
+                String policyJsonTemplate = getResourceFileAsString("definitions/" + policyFileName);
+                if (policyJsonTemplate != null) {
+                    String policyJson = policyJsonTemplate
+                                            .replace("${DATALAKE_BUCKET}", datalakeBucket)
+                                            .replace("${STORAGE_LOCATION_BASE}", storageLocationBase)
+                                            .replace("${DYNAMODB_TABLE_NAME}", s3GuardDynamoDBTableName);
+                    Policy policy = Policy.fromJson(policyJson);
+                    for (Statement statement : policy.getStatements()) {
+                        Set<String> actions = statement.getActions().stream().map(Action::getActionName).collect(Collectors.toCollection(TreeSet::new));
+                        Set<String> resources = statement.getResources().stream().map(Resource::getId).collect(Collectors.toCollection(TreeSet::new));
+                        for (Role role : roles) {
+                            SimulatePrincipalPolicyRequest simulatePrincipalPolicyRequest = new SimulatePrincipalPolicyRequest();
+                            simulatePrincipalPolicyRequest.setPolicySourceArn(role.getArn());
+                            simulatePrincipalPolicyRequest.setActionNames(actions);
+                            simulatePrincipalPolicyRequest.withResourceArns(resources);
+                            SimulatePrincipalPolicyResult simulatePrincipalPolicyResult =
+                                iam.simulatePrincipalPolicy(simulatePrincipalPolicyRequest);
+                            simulatePrincipalPolicyResult.getEvaluationResults().stream()
+                                .filter(evaluationResult -> evaluationResult.getEvalDecision().toLowerCase().contains("deny"))
+                                .map(evaluationResult -> String.format("%s:%s:%s", role.getArn(),
+                                    evaluationResult.getEvalActionName(), evaluationResult.getEvalResourceName()))
+                                .forEach(failedActions::add);
+                        }
+                    }
+                }
+            } catch (IOException e) {
+                LOGGER.debug("Unable to load policy json", e);
+            }
+        }
+        return failedActions;
+    }
+
+    private void checkIDBrokerAssumeRolePermissions(InstanceProfile instanceProfile,
```

Comment:
Can you please create separate classes for each of these methods, and unit tests them separately? These are pretty complex and this class is getting too large with it's 300 lines.

Reply:
Yea I have this mostly refactored locally. Didn't push the latest yet. Will comment when finished refactoring into separate classes.

Reply:
Alrighty, thanks for the update!

Reply:
Just pushed with all the testing changes.